mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -146,7 +146,7 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
  * This is a temporary fix for the case when target labels do not exist during prediction.
  */
 @Since("1.4.0")
-class StringIndexerModel (
+class StringIndexerModel private[ml] (
```

Comment:
Hmm, making this private now will break user code if it is used, no? It's unfortunate that it wasn't private before, but do we want to do this?